src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,558 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DetectPreviewFeatureAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.PreferConstCharOverConstUnitStringFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DetectPreviewFeatureAnalyzer,
+    Microsoft.NetCore.Analyzers.Runtime.PreferConstCharOverConstUnitStringFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class DetectPreviewFeatureUnitTests
+    {
+        private const string setupPreviewAttribute = @"
+    [AttributeUsage(AttributeTargets.Assembly |
+                AttributeTargets.Module |
+                AttributeTargets.Class |
+                AttributeTargets.Struct |
+                AttributeTargets.Delegate |
+                AttributeTargets.Interface |
+                AttributeTargets.Enum |
+                AttributeTargets.Constructor |
+                AttributeTargets.Method |
+                AttributeTargets.Property |
+                AttributeTargets.Field |
+                AttributeTargets.Event, Inherited = false)]
+    public sealed class RequiresPreviewFeaturesAttribute : Attribute
+    {
+        public RequiresPreviewFeaturesAttribute() { }
+    }
+
+";
+
+        [Fact]
+        public async Task TestMethodInvocation_Simple()
+        {
+            var csInput = @" 
+using System;
+namespace Preview_Feature_Scratch
+{" + setupPreviewAttribute +
+@"
+
+    public class Program
+    {
+        [RequiresPreviewFeatures]
+        public virtual void PreviewMethod()
+        {
+
+        }
+
+        static void Main(string[] args)
+        {
+            var prog = new Program();
+            [|prog.PreviewMethod()|];
+        }
```

Comment:
A few extra cases to consider:

- Attribute on a user-defined operator (`+`) for example, and call-site does `prog1 + prog2`.
- Attribute on a type (class/struct/interface), and call-site does some cast to this class, or a method receives this type as parameter.
- Attribute on a type, and call-site does `new MyType[] { }`
- Attribute on a class derives from `Exception, and call-site does `catch (ThisException e)`
- Attribute on a `Deconstruct` method, and call-site use deconstruction.

Reply:
All good suggestions :) I called out in the PR description that more test cases will be added. The exact cases we want to tag with the analyzer are not defined properly yet. For ex:

``` csharp
// We already have 
public struct Int32: ...
{
}
```
 Now let's say we define a new `public static interface IAddable` type
``` csharp
[RequiresPreviewFeature]
public static interface IAddable
{ }

public struct Int32: IAddable
{

}

Int32 foo = new Int32(5);
```

We don't want to tag `foo` here with a diagnostic, so we're still discussing how to best approach this. So I expect that the analyzer implementation and test cases will evolve as the rules around `RequiresPreviewFeatures` become more defined.